File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> all = new HashSet<Integer>();
        for (int n : nums) {
            all.add(n);
        }
        Map<Integer, Integer> starts = new HashMap<Integer, Integer>();
        for (int n : all) {
            if (!all.contains(n-1)) {
                starts.put(n, 1);
            }
        }
        int max = 0;
        for (int n : starts.keySet()) {
            int count = 1;
            while (all.contains(n+1)) {
                count++;
                n++;
            }
            if (count > max) {max = count;}
        }
        return max;
    }
}

File: Data Structures & Algorithms/number-of-one-bits/submission-0.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        mask = 1
        count = 0
        for i in range(32):
            if mask > n:
                break

            if (mask & n) > 0:
                count += 1
            mask = mask << 1
        return count